docs/architecture/adr-074-timeout-params.md

```diff
@@ -0,0 +1,231 @@
+they have similar bandwidth and machine resources. The timings for propagating votes
+and blocks are likely to be reasonably similar across networks. As a result, the
+timeout parameters are good candidates for being hardcoded. Hardcoding the timeouts
+in Tendermint would mean entirely removing these parameters from any configuration
+that could be altered by either an application or a node operator. Instead,
+Tendermint would ship with a set of timeouts and all applications using Tendermint
+would use this exact same set of values.
+
+While Tendermint nodes often run with similar bandwidth and on similar cloud-hosted
+machines, there are enough points of variability to make configuring
+consensus timeouts meaningful. Namely, Tendermint network topologies are likely to be
+very different from chain to chain. Additionally, applications may vary greatly in 
+how long the `Commit` phase may take. Applications that perform more work during `Commit`
+require a longer `TimeoutCommit` to allow the application to complete its work
+and be prepared for the next height.
+
+* From https://github.com/tendermint/spec/issues/359 it looks like `TimeoutCommit` really only exists for the purpose of allowing
+precommits to be collected after the +2/3 has initially been seen. Not much more value to it than that.
+
+* allowing long timeout commit discussed here: https://github.com/tendermint/tendermint/issues/5911#issuecomment-973560381
+
+## Decision
+
+None
+
+## Detailed Design
+
+### New Consensus Parameters
+
+A new `TimeoutParams` `message` will be added to the [params.proto file][consensus-params-proto].
+This message will have the following form:
+
+```proto
+message TimeoutParams {
+ google.protobuf.Duration propose = 1;
+ google.protobuf.Duration propose_delta = 2;
+ google.protobuf.Duration vote = 3;
+ google.protobuf.Duration vote_delta = 4;
+ google.protobuf.Duration commit = 5;
+ bool commit_timeout_bypass_enabled = 6;
+}
+```
+
+This new message will be added as a field into the [`ConsensusParams`
+message][consensus-params-proto]. The same default values that are [currently
+set for these parameters][current-timeout-defaults] in the local configuration
+file will be used as the defaults for these new consensus parameters in the
+[consensus parameter defaults][default-consensus-params].
+
+The new consensus parameters will be subject to the same
+[validity rules][time-param-validation] as the current configuration values,
+namely, each value must be non-negative.
+
+### Migration
+
+Migration to using these new parameters will occur across two releases.
+During the v0.36 release, Tendermint will add these as consensus parameters.
+The old parameters config-local parameters will continue to exist as well in the
+v0.36 release. If the parameters are set as consensus parameters, the config-local
+parameters will be ignored. If the consensus parameters are not set, Tendermint
+will fall back to using the values contained in the `config.toml` file.
```

Comment:
Does this mean that we are allowing networks to ignore the consensus params if they want and still use every nodes local config parameters. Are we sure we want to do this?

Reply:
Once the network sets a non-default values in the consensus parameters, we will use those. This just gives a release during which networks can make the upgrade.

I think that this is a good way for networks to gracefully upgrade with minimal disruption. The values in the config file are strictly fallbacks in case no values are set by the chain. It maintains a partial version of the already kind of bad thing for a release so that we can more seamlessly transition to better management of these params.